vdr/vdr.go

```diff
@@ -69,6 +70,57 @@ type VDR struct {
 	eventManager      events.Event
 }
 
+func (r *VDR) DeriveWebDIDDocument(ctx context.Context, baseURL url.URL, nutsDID did.DID) (*did.Document, error) {
+	nutsDIDDocument, _, err := r.Resolver().Resolve(nutsDID, nil)
+	if err != nil {
+		return nil, err
+	}
+	isOwner, err := r.IsOwner(ctx, nutsDID)
+	if err != nil {
+		return nil, err
+	}
+	if !isOwner {
+		log.Logger().
+			WithError(err).
+			WithField(core.LogFieldDID, nutsDID).
+			Info("Tried to derive did:web document from Nuts DID that is not owned by this node")
+		return nil, types.ErrNotFound
+	}
+
+	// Replace did:nuts DIDs with did:web, but only for owned DIDs
+	ownedNutsDIDs := []did.DID{nutsDID}
+	for _, controller := range nutsDIDDocument.Controller {
+		for _, curr := range ownedNutsDIDs {
+			if curr.String() == controller.String() {
+				continue
+			}
+		}
+		if controller.Method == didnuts.MethodName && !controller.IsURL() {
+			isOwner, err := r.IsOwner(ctx, controller)
+			if err != nil {
+				return nil, err
+			}
+			if isOwner {
+				ownedNutsDIDs = append(ownedNutsDIDs, controller)
+			}
+		}
+	}
```

Comment:
[Q] The Controller is authorized to make changes to a DID, what does this mean on a did document that has no history like did:web?

We can verify the history through the linked did:nuts, but perhaps the Controller should be deleted from did:web?